frontend/src/metabase/dashboard/components/Dashboard/use-dashboard-scroll.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from "react";
+
+import { getMainElement } from "metabase/lib/dom";
+
+export const useDashboardScroll = ({
```

Comment:
nit: might be nice to add a docstring here on what this hook does, it's not immediately obvious from the method name. Also, perhaps changing the hook name to `useHasDashboardScroll` might help?